src/test/java/com/google/api/codegen/GapicTestBase.java

```diff
@@ -0,0 +1,197 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen;
+
+import com.google.api.Service;
+import com.google.api.codegen.MultiYamlReader;
+import com.google.api.codegen.gapic.GapicProvider;
+import com.google.api.codegen.gapic.GapicProviderFactory;
+import com.google.api.tools.framework.model.Diag;
+import com.google.api.tools.framework.model.DiagCollector;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.stages.Merged;
+import com.google.api.tools.framework.model.testing.ConfigBaselineTestCase;
+import com.google.api.tools.framework.snippet.Doc;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.truth.Truth;
+import com.google.protobuf.Message;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+/**
+ * Base class for code generator baseline tests.
+ */
+public abstract class GapicTestBase extends ConfigBaselineTestCase {
+
+  private static final Pattern BASELINE_PATTERN = Pattern.compile("((?:-|\\w)+)\\[((?:-|\\w)+)\\]");
```

Comment:
Yes, it does come from GeneratorTestBase, but I do think an example would be helpful :-)